src/app/app.module.ts

```diff
@@ -5,20 +5,13 @@ import { BrowserAnimationsModule } from '@angular/platform-browser/animations'
 import { AppRoutingModule } from './app-routing.module'
 import { AppComponent } from './app.component'
 import { SharedModule } from './shared/shared.module'
-import { HttpClientModule } from '@angular/common/http'
+import { provideHttpClient, withInterceptorsFromDi } from '@angular/common/http'
 import { CoreModule } from './core/core.module'
 
 @NgModule({
   declarations: [AppComponent],
-  imports: [
-    BrowserModule,
-    HttpClientModule,
-    AppRoutingModule,
-    BrowserAnimationsModule,
-    SharedModule,
-    CoreModule,
-  ],
-  providers: [],
   bootstrap: [AppComponent],
+  imports: [BrowserModule, AppRoutingModule, BrowserAnimationsModule, SharedModule, CoreModule],
+  providers: [provideHttpClient(withInterceptorsFromDi())],
```

Comment:
Why do you use withIncerceptorsFromDi?
https://angular.dev/api/common/http/withInterceptorsFromDi
Also, do we need interceptors at all?